test/interval_test.exs

```diff
@@ -30,6 +88,11 @@ defmodule IntervalTests do
     assert ["15:00", "15:10", "15:20", "15:30", "15:40", "15:50"] == steps
   end
 
+  test "halts when enumerating with an invalid step unit" do
+    interval = %Interval{from: ~D[2017-04-02], until: ~D[2017-05-02], step: [invalid_step: 1]}
+    assert 0 == Enum.count(interval)
```

Comment:
I guess it could be already presented `Timex.Interval.FormatError` with specified message. For example, `message: "Invalid step option"`.